gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/WsAndHttpChannelizerHandler.java

```diff
@@ -62,28 +63,35 @@ public void configure(final ChannelPipeline pipeline) {
     public void channelRead(final ChannelHandlerContext ctx, final Object obj) {
         final ChannelPipeline pipeline = ctx.pipeline();
         if (obj instanceof HttpMessage && !WebSocketHandlerUtil.isWebSocket((HttpMessage)obj)) {
-            // if the message is for HTTP and not websockets then this handler injects the endpoint handler in front
-            // of the HTTP Aggregator to intercept the HttpMessage. Therefore the pipeline looks like this at start:
+            // If the message is for HTTP and not WS then this handler injects the HTTP user-agent and endpoint handlers
+            // in front of the HTTP aggregator to intercept the HttpMessage.
+            // This replaces the WS server protocol handler so that the pipeline initially looks like this:
             //
             // IdleStateHandler -> HttpResponseEncoder -> HttpRequestDecoder ->
             //    WsAndHttpChannelizerHandler -> HttpObjectAggregator ->
+            //    WebSocketServerProtocolHandler ->
```

Comment:
Was `WebSocketServerProtocolHandler` mistakenly left out of the original comment?

Reply:
Think so, this is the pipeline request handler that gets removed when an http message comes in